bazel/setup_clang.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+# This script is adapted from https://github.com/envoyproxy/envoy/blob/main/bazel/setup_clang.sh.
+
+BAZELRC_FILE="${BAZELRC_FILE:-$(bazel info workspace)/clang.bazelrc}"
+
+LLVM_PREFIX=$1
+
+if [[ ! -e "${LLVM_PREFIX}/bin/llvm-config" ]]; then
+  echo "Error: cannot find llvm-config in ${LLVM_PREFIX}."
+  exit 1
+fi
+
+PATH="$("${LLVM_PREFIX}"/bin/llvm-config --bindir):${PATH}"
+export PATH
+
+RT_LIBRARY_PATH="$(dirname "$(find "$(llvm-config --libdir)" -name libclang_rt.ubsan_standalone_cxx-x86_64.a | head -1)")"
+
+echo "# Generated file, do not edit. If you want to disable clang, just delete this file.
+build:clang --action_env='PATH=${PATH}'
+build:clang --action_env=CC=clang
+build:clang --action_env=CXX=clang++
+build:clang --action_env='LLVM_CONFIG=${LLVM_PREFIX}/bin/llvm-config'
+build:clang --repo_env='LLVM_CONFIG=${LLVM_PREFIX}/bin/llvm-config'
+build:clang --linkopt='-L$(llvm-config --libdir)'
+build:clang --linkopt='-Wl,-rpath,$(llvm-config --libdir)'
+build:clang --define clang_cl=1
```

Comment:
This is the place that I diverts from the envoy upstream. the `clang_cl` is used in the boost config_settings option. Feel there could be a better way. @lizan

Reply:
clang_cl is the Windows option, do we use that?

Reply:
We don't. I only need a way to be able to defer that it's `--config=clang`. figured it out. Changed the BUILD.boost config_settings to use `value: {'action_env': 'CXX=clang++'}`